packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -0,0 +1,540 @@
+  int position = 0;
+  int ret = player_get_play_position(player_, &position);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] player_get_play_position failed: %s",
+              get_error_message(ret));
+  }
+  return position;
+}
+
+void VideoPlayer::Dispose() {
+  LOG_INFO("[VideoPlayer] dispose player");
+  is_initialized_ = false;
+  event_sink_ = nullptr;
+  event_channel_->SetStreamHandler(nullptr);
+
+  if (player_) {
+    player_unprepare(player_);
+    player_unset_media_packet_video_frame_decoded_cb(player_);
+    player_unset_buffering_cb(player_);
+    player_unset_completed_cb(player_);
+    player_unset_interrupted_cb(player_);
+    player_unset_error_cb(player_);
+    player_destroy(player_);
+    player_ = nullptr;
+  }
+}
+
+void VideoPlayer::SetupEventChannel(flutter::BinaryMessenger *messenger) {
+  LOG_INFO("[VideoPlayer] setup event channel");
+  std::string name =
+      "flutter.io/videoPlayer/videoEvents" + std::to_string(player_id_);
+  auto channel =
+      std::make_unique<flutter::EventChannel<flutter::EncodableValue>>(
+          messenger, name, &flutter::StandardMethodCodec::GetInstance());
+  // SetStreamHandler be called after player_prepare,
+  // because initialized event will be send in listen function of event channel
+  auto handler = std::make_unique<
+      flutter::StreamHandlerFunctions<flutter::EncodableValue>>(
+      [&](const flutter::EncodableValue *arguments,
+          std::unique_ptr<flutter::EventSink<flutter::EncodableValue>> &&events)
+          -> std::unique_ptr<
+              flutter::StreamHandlerError<flutter::EncodableValue>> {
+        LOG_INFO("[VideoPlayer] call listen of StreamHandler");
+        event_sink_ = std::move(events);
+        Initialize();
+        return nullptr;
+      },
+      [&](const flutter::EncodableValue *arguments)
+          -> std::unique_ptr<
+              flutter::StreamHandlerError<flutter::EncodableValue>> {
+        LOG_INFO("[VideoPlayer] call cancel of StreamHandler");
+        event_sink_ = nullptr;
+        return nullptr;
+      });
+  channel->SetStreamHandler(std::move(handler));
+  event_channel_ = std::move(channel);
+}
+
+void VideoPlayer::Initialize() {
+  player_state_e state;
+  int ret = player_get_state(player_, &state);
+  if (ret == PLAYER_ERROR_NONE) {
+    LOG_INFO("[VideoPlayer] player state: %d", state);
+    if (state == PLAYER_STATE_READY && !is_initialized_) {
+      SendInitialized();
+    }
+  } else {
+    LOG_ERROR("[VideoPlayer] player_get_state failed: %s",
+              get_error_message(ret));
+  }
+}
+
+void VideoPlayer::SendInitialized() {
+  if (!is_initialized_ && !is_interrupted_ && event_sink_ != nullptr) {
+    int duration = 0;
+    int ret = player_get_duration(player_, &duration);
+    if (ret != PLAYER_ERROR_NONE) {
+      LOG_ERROR("[VideoPlayer] player_get_duration failed: %s",
+                get_error_message(ret));
+      event_sink_->Error("player_get_duration failed");
```

Comment:
Also please fix all other occurrences of `event_sink_->Error()`. `get_error_message` should be enough for these cases.

```suggestion
      event_sink_->Error("player_get_duration failed", get_error_message(ret));
```

Reply:
fixed